diff -ur tensorflow/python/data/ops/multi_device_iterator_ops.py tensorflow.mod/python/data/ops/multi_device_iterator_ops.py
--- tensorflow/python/data/ops/multi_device_iterator_ops.py	2019-07-03 11:24:01.000000000 -0700
+++ tensorflow.mod/python/data/ops/multi_device_iterator_ops.py	2019-07-03 11:27:32.000000000 -0700
@@ -264,8 +264,12 @@
     # Create the per device iterators.
     self._device_iterators = []
     for i, device in enumerate(self._devices):
-      with ops.device(device):
-        ds = self._create_device_dataset(i)
+      # DAI: We need to disable per device iterators and prefetching because
+      # they currently prevent dataset iterators from being serialized and
+      # deserialized. Opt for the simplest patch which will redundantly
+      # fetch input to the CPU (_source_device) for each device.
+      with ops.device(self._source_device):  # DAI: disable per device iterators
+        ds = self._dataset  # DAI: disable prefetch
         if context.executing_eagerly():
           self._device_iterators.append(dataset_ops.make_one_shot_iterator(ds))
         else:
