diff --git a/tb_plugin/torch_tb_profiler/plugin.py b/tb_plugin/torch_tb_profiler/plugin.py
index fa15705..22b8c10 100644
--- a/tb_plugin/torch_tb_profiler/plugin.py
+++ b/tb_plugin/torch_tb_profiler/plugin.py
@@ -311,8 +311,17 @@ class TorchProfilerPlugin(base_plugin.TBPlugin):
                             # Use threading to avoid UI stall and reduce data parsing time
                             t = threading.Thread(target=self._load_run, args=(run_dir,))
                             t.start()
+
+                    if not self._is_active:
+                        self._is_active = False
+                        self._is_active_initialized_event.set()
+
                 except Exception as ex:
                     logger.warning("Failed to scan runs. Exception=%s", ex, exc_info=True)
+                finally:
+                    if not self._is_active:
+                        self._is_active = False
+                        self._is_active_initialized_event.set()
 
                 time.sleep(consts.MONITOR_RUN_REFRESH_INTERNAL_IN_SECONDS)
         except:
